### ISSUE TITLE
Can I add some more rules for laravel 9/PHP 8.1 application ?

### ISSUE THREAD

**Issue by PetroGromovo** [squizlabs/PHP_CodeSniffer#3650]
Using squizlabs/php_codesniffer 3.7.1 in php project with PhpStorm 2021.1.4
I selected PSR12 in laravel 9/PHP 8.1 application :
https://prnt.sc/zme0Neh817Cs

I would like to add some more rules, like :
1) Warning notice when in php/js/css code there are more 1 empty lines.

2) Warning notice on vars definitions: 
`$filter_status   = $request->filter_status ?? '';
`OR 
`public function details($task_id)`

But valid lines :
`$filterStatus   = $request->filter_status ?? '';
...
public function details($taskId)
`

3) Warning notice on method definition/calling 
`$this->ReadProducts();`

But valid lines :
`$this->readProducts();`

4) in JS not to use ‘;’ in the end of the line

5) As I use components, I got warning of unknown tag on line like : 
`<x-app-layout>`

How can I add such rules?

Thanks in advance!

**Comment by jrfnl** [comment]
Sounds like you want to use a custom ruleset.

In it's most basic form this would look something like this:
```xml
<?xml version="1.0"?>
<ruleset name="Project name">
	<rule ref="PSR12"/>
</ruleset>
```

If you place the file in your project root and name the `phpcs.xml.dist`, it will automatically be picked up when running PHPCS (you may need to remove the "PSR12" in your IDE settings/may need to point your IDE to the file)i.

You can then add additional rules, as well as directives for which files you want to be scanned/excluded etc in the ruleset.

Have a look at the wiki to see all the things you can do with a custom ruleset:
* https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-Ruleset
* https://github.com/squizlabs/PHP_CodeSniffer/wiki/Customisable-Sniff-Properties
